In [1]:
import numpy as np
import pandas as pd

# Identify:

Please fill-in your ID in the following cell, **only id numbers, no names!**


In [2]:
id1 = "318465291"


In [3]:
df = pd.DataFrame([id1])

Uncomment in case two students are doing the assignment together:

In [4]:
#df = pd.DataFrame([id1,id2])

In [5]:
df.to_clipboard(index=False,header=False)

Please fill your names here:

Daniel tugendhaft

# Q1

In [6]:
bs_file = 'bird strikes.csv'

In [7]:
bs_df = pd.read_csv(bs_file)

In [8]:
bs_df.columns

Index(['year', 'month', 'warning', 'visibility', 'height', 'speed',
       'species name', 'injuries', 'aircraft damage'],
      dtype='object')

In [9]:
bs_df.iloc[10:15,[0,2]]

year warning
10  1990     NaN
11  1990      No
12  1990      No
13  1990     NaN
14  1990     NaN

# Q2

In [10]:
url = 'https://raw.githubusercontent.com/nlihin/data-analytics/main/datasets/aircraft%20wildlife%20strikes.csv'

In [11]:
aws_df = pd.read_csv(url)

/var/folders/1n/l8xnr__x38x4jqs4jb3_b1_w0000gp/T/ipykernel_7488/685132185.py:1: DtypeWarning: Columns (6,12,14) have mixed types. Specify dtype option on import or set low_memory=False.
  aws_df = pd.read_csv(url)


In [12]:
aws_df.columns

Index(['Incident Year', 'Incident Month', 'Operator', 'Aircraft',
       'Aircraft Type', 'Aircraft Make', 'Aircraft Model', 'Aircraft Mass',
       'Engine Make', 'Engine Model', 'Engines', 'Engine Type',
       'Engine1 Position', 'Engine2 Position', 'Engine3 Position',
       'Engine4 Position', 'Airport', 'State', 'Warning Issued',
       'Flight Phase', 'Visibility', 'Precipitation', 'Height', 'Speed',
       'Distance', 'Species Name', 'Species Quantity', 'Flight Impact',
       'Fatalities', 'Injuries', 'Aircraft Damage', 'Radome Strike',
       'Radome Damage', 'Windshield Strike', 'Windshield Damage',
       'Nose Strike', 'Nose Damage', 'Engine1 Strike', 'Engine1 Damage',
       'Engine2 Strike', 'Engine2 Damage', 'Engine3 Strike', 'Engine3 Damage',
       'Engine4 Strike', 'Engine4 Damage', 'Engine Ingested',
       'Propeller Strike', 'Propeller Damage', 'Wing or Rotor Strike',
       'Wing or Rotor Damage', 'Fuselage Strike', 'Fuselage Damage',
       'Landing Gear Strike'

In [13]:
aws_df.loc[[10,11,25],['Species Name', 'Flight Phase', 'Aircraft']]

Species Name Flight Phase Aircraft
10   UNKNOWN SMALL BIRD        CLIMB    MD-88
11  UNKNOWN MEDIUM BIRD  TAKEOFF RUN    MD-80
25        HOUSE SPARROW  TAKEOFF RUN    C-21A

# Q3

In [14]:
len(aws_df['Aircraft'].unique())

559

In [15]:
aws_df['Aircraft'].unique()[0]

'UNKNOWN'

#without 'UNKNOWN'

In [16]:
len(aws_df['Aircraft'].unique())-1

558

The answer is 558 types.

# Q4

In [17]:
pd.DataFrame(aws_df[aws_df.loc[:,'Aircraft']=='A-320'].Height.describe())

Height
count   4099.000000
mean     874.820200
std     1891.850931
min        0.000000
25%        0.000000
50%       50.000000
75%      800.000000
max    15000.000000

The average height is: 874.820200
The median height is: 50.000000

*The difference indicates many injuries at high altitude, which creates a gap between the average and the median.*

# Q5

In [18]:
aws_df[aws_df.loc[:,'Incident Year']>=2011]

Incident Year  Incident Month                Operator        Aircraft  \
77591            2011               1             HORIZON AIR     DHC8 DASH 8   
77592            2011               1    GREAT LAKES AIRLINES           BE-19   
77593            2011               1    GREAT LAKES AIRLINES         BE-1900   
77594            2011               1    GREAT LAKES AIRLINES         BE-1900   
77595            2011               1                BUSINESS  GULFSTREAM 200   
...               ...             ...                     ...             ...   
135003           2015               9      SOUTHWEST AIRLINES       B-737-700   
135004           2015               9      SOUTHWEST AIRLINES       B-737-700   
135005           2015               9                BUSINESS     CITATIONJET   
135006           2015               9             HORIZON AIR     DHC8 DASH 8   
135007           2015               9  AIR WISCONSIN AIRLINES      CRJ100/200   

       Aircraft Type Aircraft Make Aircraft Model  Aircraft Mass  Engine Make  \
77591              A           303           10.0            3.0         31.0   
77592              A           123            3.0            1.0          7.0   
77593              A           123           27.0            3.0         31.0   
77594              A           123           27.0            3.0         31.0   
77595              A           417            NaN            3.0         31.0   
...              ...           ...            ...            ...          ...   
135003             A           148             42            4.0         10.0   
135004             A           148             42            4.0         10.0   
135005             A           226             49            2.0         46.0   
135006             A           303             10            3.0         31.0   
135007             A           188             10            3.0         22.0   

       Engine Model  ...  Fuselage Strike Fuselage Damage Landing Gear Strike  \
77591            10  ...                0               0                   0   
77592            10  ...                0               0                   0   
77593             4  ...                1               0                   0   
77594             4  ...                0               0                   0   
77595            14  ...                0               0                   0   
...             ...  ...              ...             ...                 ...   
135003            1  ...                0               0                   0   
135004            1  ...                0               0                   0   
135005            1  ...                1               1                   0   
135006           10  ...                1               0                   0   
135007            4  ...                0               0                   0   

        Landing Gear Damage Tail Strike  Tail Damage Lights Strike  \
77591                     0           0            0             0   
77592                     0           0            0             0   
77593                     0           0            0             0   
77594                     0           0            0             0   
77595                     0           0            0             0   
...                     ...         ...          ...           ...   
135003                    0           0            0             0   
135004                    0           0            0             0   
135005                    0           0            0             0   
135006                    0           0            0             0   
135007                    0           0            0             0   

       Lights Damage Other Strike Other Damage  
77591              0            0            0  
77592              0            0            0  
77593              0            0            0  
77594              0            0            0  
775

The answer is 57417 incidents.

# Q6

In [19]:
aws_df_2014 = aws_df[aws_df.loc[:,'Incident Year']==2014]

In [20]:
len(aws_df_2014[aws_df_2014.loc[:,'Aircraft']=='A-320'])

623

The answer is 623 incidents.

# Q7

In [27]:
aws_df_D23_avr = aws_df[(aws_df.Distance == 23) & (aws_df.Speed >=0)]
pd.DataFrame(aws_df_D23_avr.Speed)

Speed
40162   250.0
42605   190.0
50944   250.0
66237   210.0
69111   230.0
107190  250.0
128155  190.0

We have 7 events so it is considered a harmonic average in this way:
7 divided by the sum of 1 divided by the speed, on each plane.

The harmonic average speed of the planes that traveled a distance of 23 miles is: 221.266561